### PR TITLE
disable combo box for startup mode on Edit screen for iBFT

### PR DESCRIPTION
**Pull request by gabi2**
bug-fix for bnc#954509  probably not yet complete therefore no changelog/new version yet.

### CODE REVIEW

File: src/modules/IscsiClientLib.rb [hunk 1]
```diff
@@ -775,29 +775,40 @@ def getCurrentNodeValues
...
+    # Checks whether iSCSI session (values provided as hash) is iBFT session
+    #
+    # @param  [Hash]      iSCSI node values as hash
+    # @return [Bool]      is iSCSI session booted from firmware?
+    #
+    def isiBFT?(node_info)
+      ret = false
+      ibft = getiBFT
+
+      if (!ibft.empty?)
```
Comment (lslezak) [inline src/modules/IscsiClientLib.rb]: Here you can merge all conditions and can get rid of the `ret` variable:

```ruby
return !ibft.empty? && ibft["iface.transport_name"] == node_info["iface.transport_name"] &&
  ibft["iface.initiatorname"] == node_info["iface.initiatorname"] &&
  ibft["node.name"] == node_info["node.name"] &&
  ibft["node.conn[0].address"] == node_info["node.conn[0].address"]
```

And I'd even refactor the compare functionality (repeating all the keys is error prone):

```ruby
def equal_nodes?(n1, n2)
  return false if n1.empty?
  
  # FIXME: maybe define a constant?
  keys = [
    "iface.transport_name",
    "iface.initiatorname",
    "node.name",
    "node.conn[0].address"
  ]
  
  keys.all? { |key| n1[key] == n2[key] }
end
```

and then in the `isiBFT?` method simply call:
```ruby
  return equal_nodes?(getiBFT, node_info)
```

File: test/getiBFT_spec.rb [hunk 2]
```diff
@@ -11,6 +11,7 @@
   describe "#getiBFT" do
     context "when filtering output of 'iscsiadm -m fw'" do
       it "returns data in form of a map " do
+        Yast::Arch.stub(:architecture).and_return("x86_64")
```
Comment (lslezak) [inline test/getiBFT_spec.rb]: Use the new rspec syntax here:

```ruby
allow(Yast::Arch).to receive(:architecture).and_return("x86_64")
```

The old syntax will stop working at some point in the future...

Comment (lslezak) [inline test/getiBFT_spec.rb]: I'd even consider enabling the verifying doubles in all tests, like here https://github.com/yast/yast-registration/blob/9cd0dfe8ecea8cc45c236a14f1c3c0a548c1718c/test/spec_helper.rb#L20-L25 .

File: src/modules/IscsiClientLib.rb [hunk 3]
```diff
@@ -1711,6 +1730,8 @@ def GetDiscoveryCmd(ip, port, use_fw: false, only_new: false)
     publish :function => :GetOffloadModules, :type => "list <string> ()"
     publish :function => :LoadOffloadModules, :type => "list <string> ()"
     publish :function => :GetDiscoveryCmd, :type => "string (string, string, map)"
+    publish :function => :getCurrentNodeValues, :type => "map <string, any>) ()"
+    publish :function => :isiBFT, :type => "boolean (map <string, any>)"
```
Comment (lslezak) [inline src/modules/IscsiClientLib.rb]: Um, does it actually work from non-Ruby code? The method name is actually `isiBFT?`. I'm not sure if ruby-bindings do some tricks here or not...

BTW the usual Ruby style is to not use the `is` prefix, `?` suffix is sufficient.

Comment (gabi2) [inline src/modules/IscsiClientLib.rb]: Thanks for review, I will rename isiBFT? to iBFT? (and publish correctly)

**Comment by mvidner** [comment]
LGTM
(the failing test is fixed in #51 )

**Pull request merged**